### DISCUSSION TITLE
Is it possible to sync bookmarks through Git repository?

### DISCUSSION THREAD

**Discussion by nutnutwin** [alefragnani/vscode-bookmarks#484] [Q&A] [answered]
Hi, I works on several devices and sync the code using Gitlab(mostly)

I cannot help but noticed that Bookmarks on DeviceA is not committed to git, thus cannot be viewed from DeviceB? 

I appreciate this feature(less intrusive), but wonder if it can be uploaded, optionally, to Git. 

Thank you very much for your time

**Comment by alefragnani** [comment, marked as answer]
Hi @nutnutwin ,

If you need to store your bookmarks with your repository, you can define the `bookmarks.saveBookmarksInProject` setting as `true`. Doing so, the bookmarks will be stored in `.vscode/bookmarks.json` file, inside your repo, so you can simply commit the changes and push to remote. When you pull the changes on other devices, the bookmarks will be there.

Just remember two things:

1. You must set `bookmarks.saveBookmarksInProject` to `true` on the other devices as well.
2. You must `pull` the changes _before_ opening the repo in VS Code OR you must reload VS Code (`Reload Window` command) after pulling, because the `.vscode/bookmarks.json` file is not monitored for changes, so the extension will not recognize the new bookmarks.

More details https://github.com/alefragnani/vscode-bookmarks#available-settings

Hope this helps

**Reply by nutnutwin** [reply]
Thank you so much!
